app/Http/Controllers/BulkEntryController.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Character;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+use Inertia\Inertia;
+
+class BulkEntryController extends Controller
+{
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Contracts\Foundation\Application|\Illuminate\Http\RedirectResponse|\Illuminate\Http\Response|\Illuminate\Routing\Redirector
+     */
+    public function store(Request $request)
+    {
+        $data = $request->validate([
+            'character_id' => "required|exists:characters,id",
+            'adventure_id' => "nullable|exists:adventures,id",
+            'start_date' => "required|date",
+            'end_date' => "nullable|date",
+            // How often a session was ran, measured in times/week
+            // (1 -> once per week, 2 -> twice per week, 0.5 -> once every 2 weeks)
+            'frequency' => "required|numeric",
```

Comment:
Allowing irrational test cases was the intention, actually, since 1/0.5 = 2, or 1/0.25 = 4. In my head we would of given them some sane choice to make rather than allowing them to input what ever number they wanted, maybe with a select menu or something